esmvaltool/diag_scripts/zmnam/zmnam.py

```diff
@@ -0,0 +1,99 @@
+"""
+;;#############################################################################
+;; Zonal mean Northern Annular Mode Diagnostics
+;; Author: Federico Serva (ISAC-CNR & ISMAR-CNR, Italy)
+;; Copernicus C3S 34a lot 2 (MAGIC)
+;;#############################################################################
+;; Description
+;;    Evaluation of stratosphere-troposphere coupling
+;;    based on EOF/PC analysis of the geopotential height field
+;; Modification history
+;;    20180512-A_serv_fe: Added output netCDFs, more use of preprocessor
+;;    20180510-A_serv_fe: Routines written.
+;;#############################################################################
+"""
+
+import yaml
+import sys
+import cartopy.crs as ccrs
+import matplotlib.pyplot as plt
+import warnings
+import iris
+import iris.plot as iplt
+import iris.quickplot as qplt
+import os
+import logging
+
+from esmvaltool.diag_scripts.shared import (group_metadata, run_diagnostic,
+                                            select_metadata, sorted_metadata)
+from esmvaltool.diag_scripts.shared.plot import quickplot
+
+logger = logging.getLogger(__name__)
+
+if not sys.warnoptions:
+    warnings.simplefilter("ignore")
+
+
+def get_input_files(cfg, index=0):
+    """Get a dictionary with input files from metadata.yml files."""
+    metadata_file = cfg['input_files'][index]
+    with open(metadata_file) as file:
+        metadata = yaml.safe_load(file)
+    return metadata
+
+
+def main(cfg):
```

Comment:
Ok, do you also mean the `get_input_files` function? Otherwise some `import`s are redundant and I should clean up (also some unused variables). In general, is `diagnostic.py` the standard template? Thanks

Reply:
Yes, there should be no need for `get_input_files` anymore and diagnostic.py is a good example of how to do it (except that it contains some unnecessary stuff for demonstration purposes).

Reply:
I cleaned up the main script from unused functions